analyzers/src/SonarAnalyzer.CSharp/Rules/NonDerivedPrivateClassesShouldBeSealed.cs

```diff
@@ -43,21 +43,30 @@ public sealed class NonDerivedPrivateClassesShouldBeSealed : SonarDiagnosticAnal
         protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
             {
-                var baseTypeDeclarationSyntax = (BaseTypeDeclarationSyntax)c.Node;
-                if (IsPrivateButNotSealedType(baseTypeDeclarationSyntax))
+                var typeDeclarationSyntax = (TypeDeclarationSyntax)c.Node;
+                if (IsPrivateButNotSealedType(typeDeclarationSyntax) && !HasVirtualMembers(typeDeclarationSyntax))
```

Comment:
sorry for the late review I forgot to publish the question

I am a bit confused because the scenario in the issue does have inheritance, so it should have been detected by the code? why wasn't it detected?

Reply:
nevermind I read better the issue

Reply:
I'm not sure if we addressed the first question here:
The check for the existence of a virtual member is indeed done at the AST level.

Reply:
from our offline discussion: the initial issue had two hidden FP causes:
- nested inside nested
- class w/ virtual members